### PR TITLE
Refactor class components to functional - from begining to namespace-detail

### PR DESCRIPTION
**Pull request by MilanPospisil**
No-Issue

### CODE REVIEW

File: src/components/about-modal/about-modal.tsx [hunk 1]
```diff
@@ -8,6 +8,7 @@ import {
   TextListVariants,
 } from '@patternfly/react-core';
 import { detect } from 'detect-browser';
+import { useEffect, useState } from 'react';
 import * as React from 'react';
```
Comment (himdel) [inline src/components/about-modal/about-modal.tsx:12]: ideally, merge these into 

```suggestion
import React, { useEffect, useState } from 'react';
```

(but we can do it in a separate PR)

File: src/components/collection-count/collection-count.tsx [hunk 2]
```diff
@@ -1,47 +1,26 @@
...
-      loading: true,
-    };
-  }
 
-  componentDidMount() {
-    this.getCollectionCount(this.props.distributionPath);
-  }
+export const CollectionCount = ({ distributionPath }: IProps) => {
+  const [collectionCount, setCollectionCount] = useState(null);
+  const [alerts, setAlerts] = useState([]);
```
Comment (himdel) [inline src/components/collection-count/collection-count.tsx:14]: if you want to preserve the `AlertType[]` check from `IState`, you can do `useState<AlertType[]>([])` here

**Review by himdel — APPROVED** [review]
LGTM, +- the linter issues :)

File: src/components/collection-count/collection-count.tsx (hunk 2 above)
Comment (MilanPospisil) [reply src/components/collection-count/collection-count.tsx:14]: Ah, classic generics :-) Thanks.

File: src/components/collection-detail/table-of-contents.tsx [hunk 3]
```diff
@@ -48,242 +44,229 @@ interface IProps {
...
+          table.roles.push(getContentEntry(content, baseUrlParams));
+          break;
+        case 'module':
+          table.modules.push(getContentEntry(content, baseUrlParams));
+          break;
+        case 'playbook':
+          table.playbooks.push(getContentEntry(content, baseUrlParams));
+          break;
+        default:
+          table.plugins.push(this.getContentEntry(content, baseUrlParams));
```
Comment (himdel) [inline src/components/collection-detail/table-of-contents.tsx:153]: `this`
